### PR TITLE
correct 'Height' when website gives 0

### LINKED ISSUE #288: When I right-click to select links, page autmatically jumps to the bottom of the page with no links selected
It only happens on one site, www.bricklink.com.  for example, at this url: https://www.bricklink.com/catalogItemInv.asp?S=75175-1&viewItemType=M I want to select the three links directly to the left of the pictures of the minifigures (sw0818, sw0820, sw0819).  When I right click, the site jumps to the very bottom of the page without selecting any links.

### PR DESCRIPTION
**Pull request by remi-garcia**
Closes #288 #263 

Some websites give `0` as `document.documentElement.scrollHeight` and `document.documentElement.clientHeight`. When that's the case I used [this](https://javascript.info/size-and-scroll-window#width-height-of-the-document) method. Except for `GetClientDims()` where I explicitly comment the reason why.

### CODE REVIEW

**Comment by remi-garcia** [comment]
This might fix #222 too

File: src/content-scripts/ElementIndexer.js [hunk 1]
```diff
@@ -339,8 +339,16 @@ class ElementIndexer {
 		if(Prefs.Debug_Measure_SearchSpeed)
 			rr = new RateReporter('Found ${Count} Elements in ${Elapsed} (${PerSecond})');
 
-		let docHeight   = document.documentElement.scrollHeight,
-			Buckets     = Prefs.IndexBuckets,
+			let docHeight = document.documentElement.scrollHeight;
```
Comment (cpriest) [inline src/content-scripts/ElementIndexer.js]: This is all duplicate code and should be put into a function, such as in globals.  Probably name it like GetDocumentDims() : [ int, int ]

File: src/content-scripts/SelectionRect.js [hunk 2]
```diff
@@ -123,8 +123,17 @@ class Rect {
...
+    let docHeight = docElem.scrollHeight;
+    // To handle some websites (#288)
+    if(docHeight == 0) {
+      docHeight = Math.max(
+        document.body.scrollHeight, docElem.scrollHeight,
+        document.body.offsetHeight, docElem.offsetHeight,
+        document.body.clientHeight, docElem.clientHeight
+      );
+    }
+    super(0, 0, docHeight, docElem.scrollWidth);
```
Comment (cpriest) [inline src/content-scripts/SelectionRect.js]: Technically if we're compensating for height peculiarities, the width should be accommodated as well.

**Comment by cpriest** [comment]
Thanks for all the help you're putting in, super appreciated!  Nice find on that post.

**Comment by remi-garcia** [comment]
I just realize there is still an issue with [bricklink](https://www.bricklink.com/catalogItemInv.asp?S=75175-1&viewItemType=M) (issue #288). Rectangle is working but that doesn't select anything. This should be resolve before merge, I'll look into it.

**Comment by cpriest** [comment]
Sounds good.  After these pull requests, please start working off the dev branch which is branched off master.  I will merge these into dev once done.

**Comment by remi-garcia** [comment]
Tests pass for me on bricklink.

> After these pull requests, please start working off the dev branch

All right.

**Comment by remi-garcia** [comment]
I changed the base branch. That added #294 to this PR.

**Comment by cpriest** [comment]
Awesome!  I'll squash & merge this to dev.  I've also been pushing quite a bit of code/changes as well so you may want to do a rebase from remotes/dev on your end.

**Comment by remi-garcia** [comment]
Before any other PR I'll try to switch for VS as you suggested and delete my fork to start over properly. Thank you very much for the time you take helping me with my PRs and issues, it's not my field so I've a lot to learn.

**Comment by cpriest** [comment]
I'm more than happy to help you with anything as well.  I love to mentor.  If you shoot me an email to my github email, I'll send you an updated email address.

I think we're in radically different time zones (France vs US CDT?) but I'm also on Gitter in the SnapLinks channel.

**Pull request merged**